### PR TITLE
WIP: Local- and ZonedDateTimeAdapter

### PR DESCRIPTION
**Pull request by astyz**
Added a Local- and ZonedDateTime converter to make it possible to use them as parameter and return type according ISO 8601 through the WebApp.
Actually needed for: #172 #173 

Pull request status:
- [x] Tests added/updated
- [x] Ready for review

### CODE REVIEW

File: plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/BestTransportResult.java [hunk 1]
```diff
@@ -119,4 +121,31 @@ public String routeToShortString() {
...
 	}
 
+	/**
+	 * @see java.lang.Object#equals
+	 */
+	@Override
+	public boolean equals(Object obj) {
+		if (obj instanceof BestTransportResult) {
+			BestTransportResult best = (BestTransportResult) obj;
+			return this.mode == best.mode && this.route.bounds == best.route.bounds
```
Comment (Legion2) [inline plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/BestTransportResult.java]: use `equals` on mode and route

File: plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/BestTransportResult.java [hunk 2]
```diff
@@ -119,4 +121,31 @@ public String routeToShortString() {
...
+		return false;
+	}
+
+	/**
+	 * @see java.lang.Object#hashCode()
+	 */
+	@Override
+	public int hashCode() {
+		// TODO Auto-generated method stub
+		return super.hashCode();
```
Comment (Legion2) [inline plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/BestTransportResult.java]: calculate hashcode from hashcode of mode and route

File: plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/NavigationResource.java [hunk 3]
```diff
@@ -78,34 +71,58 @@
...
 	 *            How to travel: Allowed parameters: driving, car, bicycling, bike, transit, public transport,
 	 *            transport, walking, walk
-	 * @param timestamp
-	 *            time object which sets the departure time, can be null
 	 * @return a data structure with the best route and the transport type of the query
 	 */
 	@POST
 	@Path("fromTo")
-	@Consumes(MediaType.APPLICATION_JSON)
+	@Consumes(MediaType.TEXT_PLAIN)
```
Comment (Legion2) [inline plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/NavigationResource.java]: there is no input

File: plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/NavigationResource.java [hunk 4]
```diff
@@ -114,25 +131,23 @@ public BestTransportResult routeFromTo(@QueryParam("origin") @DefaultValue("") S
...
 	 *            time you plan to arrive at the destination
-	 * @return a Timestamp object with the latest start time
+	 * @return ZonedDateTime object with the latest start time
 	 */
 	@POST
-	@Path("when")
-	@Consumes(MediaType.APPLICATION_JSON)
+	@Path("when/{arrivalTime}")
+	@Consumes(MediaType.TEXT_PLAIN)
 	@Produces(MediaType.APPLICATION_JSON)
```
Comment (Legion2) [inline plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/NavigationResource.java]: this should be TEXT_PLAIN

File: plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/rest/Timestamp.java [hunk 5]
```diff
@@ -23,17 +23,40 @@
...
+import de.unistuttgart.iaas.amyassist.amy.utility.rest.adapter.ZonedDateTimeAdapter;
 
 /**
  * Timestamp class serves as entity for time and date
  * 
  * @author Muhammed Kaya
  */
 @XmlRootElement
-public class Timestamp extends Entity{
+public class Timestamp extends Entity {
```
Comment (Legion2) [inline plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/rest/Timestamp.java]: is Timestamp still needed?

File: http-server/src/main/java/de/unistuttgart/iaas/amyassist/amy/httpserver/adapter/LocalDateTimeMessageBodyWriter.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	/**
+	 * @see javax.ws.rs.ext.MessageBodyWriter#writeTo(java.lang.Object, java.lang.Class, java.lang.reflect.Type,
+	 *      java.lang.annotation.Annotation[], javax.ws.rs.core.MediaType, javax.ws.rs.core.MultivaluedMap,
+	 *      java.io.OutputStream)
+	 */
+	@Override
+	public void writeTo(LocalDateTime t, Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType,
+			MultivaluedMap<String, Object> httpHeaders, OutputStream entityStream)
+			throws IOException {
+		Writer writer = new PrintWriter(entityStream);
```
[the remark this answers is not in the document]
Comment (Legion2) [reply http-server/src/main/java/de/unistuttgart/iaas/amyassist/amy/httpserver/adapter/LocalDateTimeMessageBodyWriter.java]: use ISO-8859-1 Encoding

File: plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/rest/Timestamp.java (hunk 5 above)
Comment (astyz) [reply plugins/navigation/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/navigation/rest/Timestamp.java]: Not needed anymore, I'll delete it.

**Pull request merged**